eth/protocols/bsc/handler.go

```diff
@@ -74,6 +86,11 @@ func MakeProtocols(backend Backend, dnsdisc enode.Iterator) []p2p.Protocol {
 // Handle is the callback invoked to manage the life cycle of a `bsc` peer.
 // When this function terminates, the peer is disconnected.
 func Handle(backend Backend, peer *Peer) error {
+	select {
+	case <-peer.handshaked:
```

Comment:
This seems not extensible, as the channel is only used by `bsc` protocol, it wont work if another extension is needed in the future.

Reply:
snap has no handshake, it's easy.
all problem begin from a sub protocol need handshake

Reply:
> This seems not extensible, as the channel is only used by `bsc` protocol, it wont work if another extension is needed in the future.


another sub protocol need handshake should have it's own `handshaked ` channel